Cargo.toml

```
@@ -30,3 +30,10 @@ hyper = { git = "https://github.com/hyperium/hyper", rev = "ed2b22a7f66899d33869
 metrics = { git = "https://github.com/ZcashFoundation/metrics", rev = "971133128e5aebe3ad177acffc6154449736cfa2" }
 metrics-exporter-prometheus = { git = "https://github.com/ZcashFoundation/metrics", rev = "971133128e5aebe3ad177acffc6154449736cfa2" }
 tower = { git = "https://github.com/tower-rs/tower", rev = "d4d1c67c6a0e4213a52abcc2b9df6cc58276ee39" }
+
+# TODO: remove these after a new librustzcash release.
+# These are librustzcash requirements specified in its workspace Cargo.toml that we must replicate here
+halo2 = { git = "https://github.com/zcash/halo2.git", rev = "236115917df9db45282fec24d1e1e36f275f71ab" }
+orchard = { git = "https://github.com/zcash/orchard.git", rev = "37b1b7f357cd34d93f9c55bb96efd05ab4e84408" }
+# Must point to the same revision used in zebra-chain/Cargo.toml
+zcash_note_encryption = { git = "https://github.com/zcash/librustzcash.git", rev = "0c3ed159985affa774e44d10172d4471d798a85a" }
```

This is a workspace patch directive, so it overrides whatever is specified in `zebra-chain/Cargo.toml` automatically:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

```suggestion
zcash_note_encryption = { git = "https://github.com/zcash/librustzcash.git", rev = "0c3ed159985affa774e44d10172d4471d798a85a" }
```

---

I think there might be an issue if the versions don't match. I.e., if `zebra-chain` points to a commit that has a semver different from the commit referenced by the patch, then the patch doesn't get applied. I'm not 100% sure of this, but I did run into errors applying a patched `halo2` because some commits referenced `v0.0.0` and some referenced `v0.0.1`, but I didn't investigate deep enough to know exactly what was happening. I just ran `cargo update -p halo2` so that they all referenced `v0.0.1` and then the patched version got applied AFAICT :man_shrugging: